### PR TITLE
Use RxJS pipeables instead of lettables

### PR DESCRIPTION
**Pull request by salemdar**
https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md

### CODE REVIEW

File: src/client/app/operators.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-// rxjs
-import 'rxjs/add/observable/throw';
-import 'rxjs/add/operator/map';
-import 'rxjs/add/operator/catch';
+// rxjs: Add only if lettable operators are being used. Pipeable operators don't require this.
```
Comment (mgechev) [inline src/client/app/operators.ts]: Would you drop this file completely? We don't want to encourage people to write imports with side effects.

**Comment by mgechev** [comment]
Left one comment only. LGTM otherwise!

File: src/client/app/operators.ts (hunk 1 above)
Comment (salemdar) [reply src/client/app/operators.ts]: Sure.

Btw, When we use `import { catchError } from 'rxjs/operators/catchError';`, it gets bundled just fine. But when we use `import { catchError } from 'rxjs/operators';`, we get all the operations in the prod bundle. There's an easy webpack solution for that, but do you think it is possible with SystemJS?

File: src/client/app/shared/name-list/name-list.service.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
-// import 'rxjs/add/operator/do';  // for debugging
+import { ErrorObservable } from 'rxjs/observable/ErrorObservable';
```
Comment (mgechev) [inline src/client/app/shared/name-list/name-list.service.ts]: Once version 6 is out we should update the imports as well.

**Pull request merged**